src/core/core.scale.js

```diff
@@ -296,10 +304,22 @@ var Scale = Element.extend({
 
 		// New implementations should return an array of objects but for BACKWARD COMPAT,
 		// we still support no return (`this.ticks` internally set by calling this method).
-		ticks = me.buildTicks() || [];
+		ticks = me.buildTicks();
 
 		// Allow modification of ticks in callback.
-		ticks = me.afterBuildTicks(ticks) || ticks;
+		if (ticks) {
```

Comment:
Why do we need to move this implementation outside of `afterBuildTicks`?

Reply:
It's not necessary, but I think it improves readability. It was confusing to me before that it was passing in a parameter that was often unused. Also, now we need to do the "support old implementations" section only in the backwards compatibility code path instead of always doing it. The previous `if` check somewhat obscured what was actually being checked and was also less performant because it was being once once per tick instead of once ever